legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/helperFunctions/tempTableHelperFunctions.pure

```diff
@@ -0,0 +1,133 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import meta::relational::functions::tempTable::*;
+import meta::relational::functions::sqlQueryToString::*;
+import meta::pure::alloy::connections::alloy::isolation::*;
+import meta::relational::metamodel::relation::*;
+import meta::relational::runtime::*;
+import meta::relational::metamodel::*;
+import meta::relational::executionPlan::*;
+import meta::pure::executionPlan::*;
+import meta::relational::commands::*;
+
+function meta::relational::functions::tempTable::sqlStringToUpdateNode(sql: String[*]): UpdateSqlExecutionNode[1]
+{
+  ^meta::relational::executionPlan::UpdateSqlExecutionNode(sqlCommands = $sql, resultType = ^VoidResultType(type=meta::pure::router::store::routing::Void));
+}
+
+function meta::relational::functions::tempTable::updateNodeToFinallyNode(updateNodes: UpdateSqlExecutionNode[*]): FinallyExecutionNode[1]
+{
+  ^meta::relational::executionPlan::FinallyExecutionNode(executionNodes= $updateNodes, resultType = ^VoidResultType(type=meta::pure::router::store::routing::Void));
+}
+
+function meta::relational::functions::tempTable::toLazyVoidRelationalResult(updateNode: UpdateSqlExecutionNode[1]): UpdateSqlExecutionNode[1]
+{
+  ^$updateNode(resultType = ^LazyVoidResultType(type=meta::pure::router::store::routing::Void));
+}
+
+function meta::relational::functions::tempTable::generateUpdateNode(s: TempTableIsolationStrategy[0..1], r: UpdateSQLQueryCommands[1], dbConfig: DbConfig[1]): UpdateSqlExecutionNode[*]
+{
+  assert($s->isNotEmpty(), 'You must specify an isolation strategy in datasourceSpecification');
```

Comment:
Having an optional parameter and immediately asserting that its not optional doesn't look correct from API perspective